Portability fixes:

- Use fcntl.h for O_RDONLY and friends.
- Only use mallinfo with glibc.

--- libdaemon/server/daemon-server.c      2014-11-29 00:07:42.000000000 +0100
+++ libdaemon/server/daemon-server.c      2014-12-28 08:22:40.985928408 +0100
@@ -14,6 +14,7 @@
 #include "daemon-server.h"
 #include "daemon-log.h"
 
+#include <fcntl.h>
 #include <dlfcn.h>
 #include <errno.h>
 #include <pthread.h>

--- lib/mm/memlock.c.orig	2016-06-07 08:00:46.557058318 +0200
+++ lib/mm/memlock.c	2016-06-07 08:01:25.453505793 +0200
@@ -150,7 +150,7 @@ static void _touch_memory(void *mem, siz
 
 static void _allocate_memory(void)
 {
-#ifndef VALGRIND_POOL
+#if !defined(VALGRIND_POOL) && defined(__GLIBC__)
 	void *stack_mem;
 	struct rlimit limit;
 	int i, area = 0, missing = _size_malloc_tmp, max_areas = 32, hblks;
